core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala

```diff
@@ -43,9 +44,14 @@ import scala.concurrent.Future
  *                                     +----------------------+
  */
 object Scheduler {
+
+  private val log = Logger.takingImplicit[KvArgs](getClass)
```

Comment:
do we have to have the `takingImplicit` there? I would go for `withContext` for example?

Reply:
`takingImplicit` is from `scalalogging` library.
One can do a 
```
 private val log = Logger.takingImplicit[KvArgs](getClass).withContext("name", "value")
```
however, but `takingImplicit` is mandatory, may be we can create a trait (just like `StrictLogging`) if you think that would be more readable ?!